experiment/dash/dash.go

```diff
@@ -3,42 +3,132 @@ package dash
+type TestKeys struct {
+	Simple `json:"simple"`
+
+	Client_resolver string `json:"client_resolver"`
+
+	// Failure is the failure string
+	Failure string `json:"failure"`
+
+	Receiver_data []neubotModel.ClientResults `json:"receiver_data"`
 }
 
 func measure(
 	ctx context.Context, sess *session.Session, measurement *model.Measurement,
 	callbacks handler.Callbacks, config Config,
 ) error {
-	settings := measurementkit.NewSettings(
-		"Dash", sess.SoftwareName, sess.SoftwareVersion,
-		sess.CABundlePath(), sess.ProbeASNString(), sess.ProbeCC(),
-		sess.ProbeIP(), sess.ProbeNetworkName(), config.LogLevel,
-	)
-	out, err := measurementkit.StartEx(settings, sess.Logger)
+	testkeys := &TestKeys{
+		Simple: Simple{
+			Connect_latency:   0.0,
+			Min_playout_delay: 0.0,
+		},
+	}
+	measurement.TestKeys = testkeys
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+	client := client.New(sess.SoftwareName, sess.SoftwareVersion)
+	channel, err := client.StartDownload(ctx)
 	if err != nil {
+		testkeys.Failure = err.Error()
 		return err
 	}
-	for ev := range out {
-		mkevent.Handle(sess, measurement, ev, callbacks)
+	callbacks.OnProgress(0, fmt.Sprintf("server: %s", client.FQDN))
+	var rates []float64
+	var frame_ready_time float64 = 0.0
+	var play_time float64 = 0.0
+	percentage := 0.00
+	step := 100 / (maxRuntime + 1) / 100
+	for results := range channel {
+		percentage += step
+		message := fmt.Sprintf(
+			"rate: %d kbit/s speed: %.2f kbit/s elapsed: %.2f s",
+			results.Rate,
+			float64(results.Received*8/1024)/results.Elapsed,
+			results.Elapsed,
+		)
+		callbacks.OnProgress(percentage, message)
+		data, err := json.Marshal(results)
+		if err != nil {
+			testkeys.Failure = err.Error()
+			return err
+		}
+		sess.Logger.Debugf("%s", string(data))
+
+		rates = append(rates, float64(results.Rate))
+		if testkeys.Simple.Connect_latency == 0.0 {
+			// It is always equal for all the records
+			testkeys.Simple.Connect_latency = results.ConnectTime
+		}
+
+		frame_ready_time += float64(results.Elapsed)
+		if play_time == 0 {
+			play_time += frame_ready_time
+		} else {
+			play_time += float64(results.ElapsedTarget)
+		}
+		var stall float64 = frame_ready_time - play_time
+		if stall > testkeys.Simple.Min_playout_delay {
+			testkeys.Simple.Min_playout_delay = stall
+		}
+		testkeys.Receiver_data = append(testkeys.Receiver_data, results)
 	}
+
+	median, err := stats.Median(rates)
+	testkeys.Simple.Median_bitrate = int64(median)
+	if err != nil {
+		testkeys.Failure = err.Error()
+		return err
+	}
+	if client.Error() != nil {
+		testkeys.Failure = err.Error()
+		return client.Error()
+	}
+	data, err := json.Marshal(client.ServerResults())
+	if err != nil {
+		testkeys.Failure = err.Error()
+		return err
+	}
+	sess.Logger.Debugf("Server result: %s", string(data))
+	callbacks.OnProgress(1, "done")
+	testkeys.Client_resolver = sess.Location.ResolverIP
+	sess.Logger.Info("Test Summary: ")
```

Comment:
Printing summary at info level is an activity reserved to `github.com/ooni/probe-cli`. We should not use the info level in this codebase, because this is part of the (currently implicit) contract with the real OONI client.